src/app/dev/DevToys.Tools/Tools/Converters/JsonTable/JsonTableConverterGuiTool.cs

```diff
@@ -0,0 +1,210 @@
+﻿using System.Data;
+using DevToys.Tools.Tools.Converters.JsonTable;
+using Microsoft.Extensions.Logging;
+using static DevToys.Tools.Helpers.JsonTableHelper;
+
+namespace DevToys.Tools.Tools.Converters.JsonYaml;
+
+[Export(typeof(IGuiTool))]
+[Name("JsonTableConverter")]
+[ToolDisplayInformation(
+    IconFontName = "DevToys-Tools-Icons",
+    IconGlyph = '\u0109',
+    GroupName = PredefinedCommonToolGroupNames.Converters,
+    ResourceManagerAssemblyIdentifier = nameof(DevToysToolsResourceManagerAssemblyIdentifier),
+    ResourceManagerBaseName = "DevToys.Tools.Tools.Converters.JsonTable.JsonTableConverter",
+    ShortDisplayTitleResourceName = nameof(JsonTableConverter.ShortDisplayTitle),
+    LongDisplayTitleResourceName = nameof(JsonTableConverter.LongDisplayTitle),
+    DescriptionResourceName = nameof(JsonTableConverter.Description),
+    AccessibleNameResourceName = nameof(JsonTableConverter.AccessibleName),
+    SearchKeywordsResourceName = nameof(JsonTableConverter.SearchKeywords))]
+[AcceptedDataTypeName(PredefinedCommonDataTypeNames.Json)]
+internal sealed partial class JsonTableConverterGuiTool : IGuiTool, IDisposable
+{
+    /// <summary>
+    /// Whether the tool should copy to clipboard as CSV or TSV.
+    /// </summary>
+    /// <remarks>
+    /// Default to tab-separated-values because copy-to-Excel is the most likely use-case.
+    /// </remarks>
+    private static readonly SettingDefinition<CopyFormat> copyFormat
+        = new(name: $"{nameof(JsonTableConverterGuiTool)}.{nameof(copyFormat)}", defaultValue: CopyFormat.TSV);
+
+    private enum GridColumn
+    {
+        Content
+    }
+
+    private enum GridRow
+    {
+        Header,
+        Content,
+    }
+
+    private readonly ILogger _logger;
+    private readonly ISettingsProvider _settingsProvider;
+    private readonly IClipboard _clipboard;
+
+    private readonly IUIMultiLineTextInput _inputTextArea = MultilineTextInput("json-input-text-area");
+    private readonly IUIDataGrid _outputDataGrid = DataGrid("output-data-grid");
+    private readonly IUIButton _copyButton = Button("copy-data-grid");
+
+    private readonly DisposableSemaphore _semaphore = new(); // one convert at a time
+    private CancellationTokenSource? _cancellationTokenSource;
+
+    [ImportingConstructor]
+    public JsonTableConverterGuiTool(ISettingsProvider settingsProvider, IClipboard clipboard)
+    {
+        _logger = this.Log();
+        _settingsProvider = settingsProvider;
+        _clipboard = clipboard;
+    }
+
+    internal Task? WorkTask { get; private set; }
+
+    public UIToolView View
+    {
+        get
+        {
+            _copyButton
+                .Icon("FluentSystemIcons", '\uF32B')
+                .OnClick(OnCopyDataGrid)
+                .Disable(); // disable until valid JSON is input
+
+            return new(
+                isScrollable: true,
```

Comment:
Let's set `isScrollable` to `false` otherwise the DataGrid and TextBox will vertically expand as much as the content gets longer.